### ISSUE TITLE
Improve streaming mechanism

### ISSUE THREAD

**Issue by harmony7** [fastly/http-compute-js#2]
The `ComputeJsServerResponse` object is an implementation of `ServerResponse` from Node.js whose `Writable` is wired to an in-memory buffer.

One of the members of this class is `computeResponse`, a `Promise` that resolves to a `Response`. Currently this object resolves in response to the `finish` event, at which point a `Response` object is instantiated, initialized with the status code, headers, and body based on the data that has been written to the in-memory buffer, and used as the resolution value.

The problem here is that the Response object isn't created until `finish`, so the response can't really be streamed. if the process performs anything that takes time, such as performing other fetches, the creation of the Response object is held off until all this finishes.

This should be changed to resolve as soon as `_headerSent` is set `true`, and then to set up the body to be a `ReadableStream` that gets fed by `_writtenDataBuffer` as its entries get set to `.written = true`.

**Comment by harmony7** [comment]
Have released `0.3.0` that does this.

With a simple example:

```javascript
/// <reference types="@fastly/js-compute" />

import http from '@fastly/http-compute-js';

const server = http.createServer(async (req, res) => {
  res.setHeader('Content-Type', 'text/plain');
  res.write('foo before\n');

  const response = await fetch('https://http-me.glitch.me/now?wait=2000', {
    backend: 'http-me',
  });
  const text = await response.text();
  res.write(text + '\n');
  res.write('foo after\n');
  res.end();
});

server.listen();
```

This code now streams
```
foo before
```
and then this comes in two seconds later:
```
2022-10-21T06:42:33.374Z
foo after
```
Compare time to first byte:

(before)
<img width="1283" alt="Screen Shot 2022-10-21 at 14 06 16" src="https://user-images.githubusercontent.com/1396801/197315442-d4055cb9-9b5e-43f7-ae4b-159c3ef08167.png">

(after)
<img width="1277" alt="Screen Shot 2022-10-21 at 14 04 49" src="https://user-images.githubusercontent.com/1396801/197315457-97ace583-f4b3-4871-b7f1-4a94fce152e0.png">

Additionally, `0.3.1` made a change where streams using UTF8 would be converted to a byte stream using the built-in `TextEncoder` instead of `Buffer.from` which was using a polyfill, greatly improving performance.

**Comment by harmony7** [comment]
Closing this one out.
